src/java.base/unix/native/criuengine/criuengine.cpp

```diff
@@ -131,12 +181,23 @@ static int checkpoint(pid_t jvm,
     pid_t child = fork();
     if (!child) {
         execv(criu, const_cast<char **>(args.data()));
-        perror("criu dump");
-        exit(1);
+        std::cerr << "Cannot execute CRIU \"" << join_args(args) << "\": " << strerror(errno) << std::endl;
+        exit(SUPPRESS_ERROR_IN_PARENT);
     }
 
     int status;
-    if (child != wait(&status) || !WIFEXITED(status) || WEXITSTATUS(status)) {
+    if (child != wait(&status)) {
+        std::cerr << "Error waiting for spawned CRIU \"" << join_args(args) << "\": " << strerror(errno) << std::endl;
+        kickjvm(jvm, -1);
+    } else if (!WIFEXITED(status)) {
+        std::cerr << "Spawned CRIU \"" << join_args(args) << "\" has not properly exited:"
+          " waitpid status %d - check " << path_abs(imagedir, log_local) << std::endl;
+        kickjvm(jvm, -1);
+    } else if (WEXITSTATUS(status)) {
+        if (WEXITSTATUS(status) != SUPPRESS_ERROR_IN_PARENT) {
+            std::cerr << "Spawned CRIU \"" << join_args(args) << "\" has not properly exited:"
+              " exit code " << WEXITSTATUS(status) << " - check " << path_abs(imagedir, log_local) << std::endl;
```

Comment:
we are not required to abs(imagedir), as unprocessed that is in the same way as it was provided in the args

Reply:
The problem is users do not execute `criuengine` themselves. From my experience:
```
$ $JTREG_HOME/bin/jtreg -jdk:$ZULU_HOME -v:all -timeout:10 $TEST_HOME/test/jdk/jdk/crac/BigCheckpointTest.java
Spawned CRIU "/home/azul/azul/crac-git/build/linux-x86_64-server-fastdebug/images/jdk/lib/criu dump -t 648896 -D cr --shell-job -v4 -o dump4.log" has not properly exited: exit code 1 - check dump4.log
$ ls -l dump4.log
ls: cannot access 'dump4.log': No such file or directory
```
How to find out `dump4.log` is hiding in the `JTwork/scratch_1/cr` subdirectory?
```
$ ls -l  ./JTwork/scratch_1/cr/dump4.log
-rw------- 1 azul azul 3547 Aug 10 06:58 ./JTwork/scratch_1/cr/dump4.log
```

Reply:
>> we are not required to abs(imagedir), as unprocessed that is in the same way as it was provided in the args
> The problem is users do not execute criuengine themselves. From my experience:
> $JTREG_HOME/bin/jtreg -jdk:$ZULU_HOME -v:all -timeout:10 $TEST_HOME/test/jdk/jdk/crac/BigCheckpointTest.java
Spawned CRIU "/home/azul/azul/crac-git/build/linux-x86_64-server-fastdebug/images/jdk/lib/criu dump -t 648896 -D cr --shell-job -v4 -o dump4.log" has not properly exited: exit code 1 - check dump4.log

My point was about absolute path for image dir. I.e. this one looks fine

> $ java -XX:CRaCheckpointTo=./cr
> Spawned CRIU "/home/azul/azul/crac-git/build/linux-x86_64-server-fastdebug/images/jdk/lib/criu dump -t 648896 -D ./cr --shell-job -v4 -o dump4.log" has not properly exited: exit code 1 - check *./cr/dump4.log* 

But Jtreg case is valid, OK.